lib/fs.js

```diff
@@ -1328,21 +1340,34 @@ FSWatcher.prototype.start = function(filename,
                                      persistent,
                                      recursive,
                                      encoding) {
+  lazyAssert()(this._handle instanceof FSEvent, 'handle must be a FSEvent');
```

Comment:
Since very recently we have the `(D)CHECK` macro. I think it would be nice to replace these assert calls with those instead :-).

Reply:
@BridgeAR Not unless `CHECK` throws instead of aborting...at least currently we only expects an error thrown in `sequential/test-fs-watch.js`

Reply:
Well... it is a semver-major PR. But this is also just a suggestion.